### PR TITLE
add test for conv2d with padding

### PR DESCRIPTION
**Pull request by Abdurrahheem**
This PR add test for conv2d with `padding=1` as the output of the convolution operation in opencv does not match with torch exactly. This causes diffusion model output deviate a little bit compared to torch(see this [PR](https://github.com/opencv/opencv/pull/25950)). This PR does not provide solution for the problem, but merely highlights it. Would it be possible to check the cause of the issue @fengyuentau @dkurt 

Test data and model for this test are located in this [PR](https://github.com/opencv/opencv_extra/pull/1196)
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by opencv-alalek** [comment]
4.x?

**Comment by dkurt** [comment]
Seems like all the tests passed. Is that expected?

**Comment by Abdurrahheem** [comment]
> Seems like all the tests passed. Is that expected?

The reason the pass is because we in opencv check if the norm of the matrices is smallaer that some tolarance values. How ever numpy as `allclose` function, which checks elements-wise distance bitween all elements and  where all elementwise distances need to be smaller that some predefined tolarance value.  See the following peace of code for referene: 


```python

class Model(nn.Module):
    def __init__(self, in_channels, model_channels, dims=2):
        super().__init__()
        self.layer = nn.Conv2d(in_channels, model_channels, kernel_size=3, padding=1)

    def forward(self, x, emb):
        output = self.layer(x)
        return output



if __name__ == "__main__":

    model = Model(32, 32)
    x = torch.randn((1, 32, 30, 30))
    emb = torch.randn((1, 10))

    output = model(x, emb).detach().cpu().numpy()

    torch.onnx.export(
        model,
        (x, emb),
        "exp.onnx",
        input_names=["x"],
        output_names=["output"]
    )

    net = cv.dnn.readNet("./exp.onnx")
    net.setInput(x.detach().cpu().numpy(), "x")

    output_cv = net.forward()

    ## let us run the onnx in onnxruntime
    sess = ort.InferenceSession("exp.onnx")
    i_name_1 = sess.get_inputs()[0].name
    output_name = sess.get_outputs()[0].name
    output_ort = sess.run([output_name], {i_name_1: x.detach().cpu().numpy()})[0]

    print("ORT match: ", np.allclose(output, output_ort))
    print("CV match: ", np.allclose(output, output_cv))
    print("torch: \n", output[0, 0, :2, :2])
    print("opencv: \n", output_cv[0, 0, :2, :2])
    print("ort: \n", output_ort[0, 0, :2, :2])

    print("diff sum opencv: ", np.sum(np.abs(output - output_cv)))
    print("diff sum ort: ", np.sum(np.abs(output - output_ort)))
    print("norm cv: ", np.linalg.norm((output - output_cv)))
    print("norm ort: ", np.linalg.norm((output - output_ort)))
```

```
ORT match:  False
CV match:  False
torch: 
 [[ 0.75934446 -0.33524114]
 [-0.57902    -0.88300264]]
opencv: 
 [[ 0.7593445  -0.33524022]
 [-0.57902116 -0.8830029 ]]
ort: 
 [[ 0.75934446 -0.33524114]
 [-0.57902    -0.8830025 ]]

diff sum opencv:  0.026568387
diff sum ort:  0.0032581482
norm cv:  0.00024282983
norm ort:  2.7956728e-05
```

This is quite a tight comparison, but some networks such as diffision are sensetinve to small deviations in the results (due to iterative nature of inference).

**Comment by fengyuentau** [comment]
The difference you show is quite small (1e-6). Does that really make a difference in the final output? Did you check if onnxruntime can produce the same output as using pytorch?

**Comment by Abdurrahheem** [comment]
> The difference you show is quite small (1e-6). Does that really make a difference in the final output? Did you check if onnxruntime can produce the same output as using pytorch?

@fengyuentau  sorry for late repsonce. 

 - Yes, it turns out that, even such a small deviation make difference in diffusion procccess. 
 - Yes, I checked diffuction model with onnxrutime, its result matches perfectly with torch (ours does not). Currenly the only not matching results are in convolution in opencv
